### PR TITLE
[Refactor] Split Management

### PR DESCRIPTION
**Pull request by andreaazzini**
Addresses #309

### CODE REVIEW

File: darwin/dataset/split_manager.py [hunk 1]
```diff
@@ -6,6 +7,51 @@
...
+                "train": Path("/path/to/split/stratified_polygon_train.txt"),
+                "val": Path("/path/to/split/stratified_polygon_val.txt"),
+                "test": Path("/path/to/split/stratified_polygon_test.txt")
+            },
+            "tag": {
+                "train": Path("/path/to/split/stratified_tag_train.txt"),
+                "val": Path("/path/to/split/stratified_tag_val.txt"),
+                "test": Path("/path/to/split/stratified_tag_test.txt")
+            }
+        }
```
Comment (Fl4m3Ph03n1x) [inline darwin/dataset/split_manager.py:45]: This is beautiful !

File: tests/darwin/dataset/split_manager_test.py [hunk 2]
```diff
@@ -15,7 +16,15 @@ def test_requires_scikit_learn():
...
+    @pytest.mark.parametrize("val_percentage,test_percentage", [(0, 0.3), (0, 0), (0.2, 0), (0.5, 0.5), (1, 0.1)])
+    def it_raises_for_invalid_split_configuration(
+        team_slug: str, team_extracted_dataset_path: Path, val_percentage: float, test_percentage: float
+    ):
+        with pytest.raises(ValueError):
+            root = team_extracted_dataset_path / team_slug / "sl"
+            split_dataset(root, release_name="latest", val_percentage=val_percentage, test_percentage=test_percentage)
+
+    @pytest.mark.parametrize("val_percentage,test_percentage", [(0.2, 0.3), (0.3, 0.2)])
     def it_should_split_a_dataset(
```
Comment (Fl4m3Ph03n1x) [inline tests/darwin/dataset/split_manager_test.py:26]: Does this test a random or stratified split?

**Review by Fl4m3Ph03n1x — COMMENTED** [review]
Looking good overall, made some minor changes and only have a question left!

File: tests/darwin/dataset/split_manager_test.py (hunk 2 above)
Comment (andreaazzini) [reply tests/darwin/dataset/split_manager_test.py:26]: That should do both!

**Review by Fl4m3Ph03n1x — APPROVED** [review]

**Pull request merged**